### PR TITLE
Optimize  ：Avoid main thread blocking

### PR DESCRIPTION
**Pull request by MentosL**
## What is the purpose of the change

Avoid main thread blocking

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/DubboBootstrap.java [hunk 1]
```diff
@@ -1273,7 +1273,7 @@ private void waitExportFinish() {
         try {
             logger.info(NAME + " waiting services exporting asynchronously...");
             CompletableFuture<?> future = CompletableFuture.allOf(asyncExportingFutures.toArray(new CompletableFuture[0]));
-            future.get();
+            future.get(3, TimeUnit.SECONDS);
```
Comment (AlbumenJ) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/DubboBootstrap.java:1276]: 3 seconds might not enough for some exporters

**Review by MentosL — COMMENTED** [review]
I think the future here should equal countdownlatch.await (). The longest export interface should also be completed within this time range.

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/DubboBootstrap.java (hunk 1 above)
Comment (MentosL) [reply dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/DubboBootstrap.java:1276]: I think the future here should equal countdownlatch.await (). The longest export interface should also be completed within this time range.

**Comment by MentosL** [comment]
Found that 3.0 has changed the file and the code is out of date

**Pull request closed without merging**